consumer/pull_consumer.go

```diff
@@ -19,40 +19,104 @@ package consumer
 
 import (
 	"context"
-	"errors"
 	"fmt"
-	"strconv"
 	"sync"
+	"sync/atomic"
 
 	"github.com/apache/rocketmq-client-go/internal"
 	"github.com/apache/rocketmq-client-go/primitive"
+	"github.com/apache/rocketmq-client-go/rlog"
+	"github.com/pkg/errors"
+)
+
+var (
+	ErrMQEmpty = errors.New("MessageQueue is nil")
+	ErrOffset  = errors.New("offset < 0")
+	ErrNumbers = errors.New("numbers < 0")
 )
 
 type PullConsumer interface {
```

Comment:
Don't re-define PullConsumer Interface, please refer `api.go`

Reply:
@wenfengwang if do like that, we will introduce the problem "import cycle", I put that in utils/errors.go.